internal/supported_types/aws_supported_types.go

```diff
@@ -0,0 +1,18 @@
+package supported_types
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+)
+
+func FilterInstanceTypes(_ context.Context, it *[]clients.InstanceType) (*[]clients.InstanceType, error) {
+	var filteredInstanceTypes []clients.InstanceType
+	for _, i := range *it {
+		_, ok := awsSupported[i.Name]
+		if ok {
+			filteredInstanceTypes = append(filteredInstanceTypes, i)
+		}
+	}
+	return &filteredInstanceTypes, nil
+}
```

Comment:
Here is a crazy idea: what if this function was declared on the instance type itself? Then you could simply write:

```go
types := client.NewInstanceTypes(resultFromAws)
types = types.FilterSupportedTypes()
```

With this syntax, I can even see that the name could be even improved to something like: `types.OnlySupported()`. Also see that context is not passed in, for trivial functions which do not need logging, passing in cotext is perhaps not needed.